editoast/src/generated_data/error/signals.rs

```diff
@@ -32,40 +32,51 @@ pub fn generate_errors(infra_cache: &InfraCache) -> Vec<InfraError> {
     for signal in infra_cache.signals().values() {
         let signal = signal.unwrap_signal();
         // Retrieve invalid refs
-        if !infra_cache.track_sections().contains_key(&signal.track) {
-            let obj_ref = ObjectRef::new(ObjectType::TrackSection, signal.track.clone());
-            let infra_error = InfraError::new_invalid_reference(signal, "track", obj_ref);
-            errors.push(infra_error);
+        let infra_errors = check_invalid_ref(signal, infra_cache);
+        if !infra_errors.is_empty() {
+            errors.extend(infra_errors);
             continue;
         }
 
-        let track_cache = infra_cache
-            .track_sections()
-            .get(&signal.track)
-            .unwrap()
-            .unwrap_track_section();
-        // Retrieve out of range
-        if !(0.0..=track_cache.length).contains(&signal.position) {
-            let infra_error = InfraError::new_out_of_range(
-                signal,
-                "position",
-                signal.position,
-                [0.0, track_cache.length],
-            );
-            errors.push(infra_error);
-        }
+        errors.extend(check_out_of_range(signal, infra_cache));
     }
 
     errors
 }
 
+/// Retrieve invalid refs for signals
+pub fn check_invalid_ref(signal: &SignalCache, infra_cache: &InfraCache) -> Vec<InfraError> {
+    if !infra_cache.track_sections().contains_key(&signal.track) {
+        let obj_ref = ObjectRef::new(ObjectType::TrackSection, signal.track.clone());
+        return vec![InfraError::new_invalid_reference(signal, "track", obj_ref)];
+    }
+    vec![]
```

Comment:
```suggestion
    if !infra_cache.track_sections().contains_key(&signal.track) {
        let obj_ref = ObjectRef::new(ObjectType::TrackSection, signal.track.clone());
        vec![InfraError::new_invalid_reference(signal, "track", obj_ref)]
    } else {
      vec![]
    }
```

You have the problem in several locations.